course7/homeworks/mathias/ls/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 3.4 FATAL_ERROR)
+
+project(ls VERSION 1.0 LANGUAGES CXX)
+
+set(Boost_USE_STATIC_LIBS ON)
+find_package(Boost COMPONENTS filesystem REQUIRED)
+
+if(MSVC)
+    add_compile_options("/permissive-" "/W4" "/WX" "$<$<CONFIG:RELEASE>:/O2>")
+else()
+    add_compile_options("-std=c++14" "-Wall" "-Wextra" "-Werror" "$<$<CONFIG:RELEASE>:-O3>")
```

Comment:
Don't specify the build type directly in the CMakeLists.txt
Allow the person actually building the project to decided whether they want release or debug build, by providing CMAKE_BUILD_TYPE in command line and configure time.

Reply:
This is a generator expression, it enables optimization on release builds only.

Reply:
Ah, OK
But I asked for for simple solution, and you couldn't resist adding all these complications :)